src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -5219,61 +5215,130 @@ private static boolean needIncrement(MutableBigInteger mdivisor, int roundingMod
         return commonNeedIncrement(roundingMode, qsign, cmpFracHalf, mq.isOdd());
     }
 
+    /**
+     * {@code FIVE_TO_2_TO[n] == 5^(2^n)}
+     */
+    private static final BigInteger[] FIVE_TO_2_TO = new BigInteger[16 + 1];
+
+    static {
+        BigInteger pow = FIVE_TO_2_TO[0] = BigInteger.valueOf(5L);
+        for (int i = 1; i < FIVE_TO_2_TO.length; i++)
+            FIVE_TO_2_TO[i] = pow = pow.multiply(pow);
+    }
+
+    /**
+     * @param n a non-negative integer
+     * @return {@code 5^(2^n)}
+     */
+    private static BigInteger fiveToTwoToThe(int n) {
+        int i = Math.min(n, FIVE_TO_2_TO.length - 1);
```

Comment:
BigInteger has “getRadixConversionCache” which could be made accessible and used here.

Reply:
I already wrote about that method previously [here](https://github.com/openjdk/jdk/pull/21323#discussion_r1790787304).

Reply:
@j3graham The benchmarks using `getRadixConversionCache()` look good (see below).
The problem with this is that the cache maintained by that method is unlimited, whereas the one controlled by `fiveToTwoToThe()` has modest memory requirements.

```
Benchmark                            Mode  Cnt          Score        Error  Units
BigDecimalStripTrailingZeros.testL   avgt   15    2723753.867 ?   9200.262  ns/op
BigDecimalStripTrailingZeros.testM   avgt   15       9269.359 ?     32.427  ns/op
BigDecimalStripTrailingZeros.testS   avgt   15        155.660 ?      0.807  ns/op
BigDecimalStripTrailingZeros.testXL  avgt   15  445079302.800 ? 969170.405  ns/op
BigDecimalStripTrailingZeros.testXS  avgt   15         21.305 ?      0.029  ns/op
Finished running test 'micro:java.math.BigDecimalStripTrailingZeros'
```

Reply:
@j3graham The benchmarks using `getRadixConversionCache()` look good (see below).
The problem with this is that the cache maintained by that method is unlimited, whereas the one controlled by `fiveToTwoToThe()` has modest memory requirements.

```
Benchmark                            Mode  Cnt          Score        Error  Units
BigDecimalStripTrailingZeros.testL   avgt   15    2723753.867 ?   9200.262  ns/op
BigDecimalStripTrailingZeros.testM   avgt   15       9269.359 ?     32.427  ns/op
BigDecimalStripTrailingZeros.testS   avgt   15        155.660 ?      0.807  ns/op
BigDecimalStripTrailingZeros.testXL  avgt   15  445079302.800 ? 969170.405  ns/op
BigDecimalStripTrailingZeros.testXS  avgt   15         21.305 ?      0.029  ns/op
Finished running test 'micro:java.math.BigDecimalStripTrailingZeros'
```

Reply:
> @j3graham The benchmarks using `getRadixConversionCache()` look good (see below). The problem with this is that the cache maintained by that method is unlimited, whereas the one controlled by `fiveToTwoToThe()` has modest memory requirements.
> 
> ```

The method here could use the radix cache up to a safe value, and continue the calculation as it currently does beyond that. The benefit would be the existing incremental evaluation.

Reply:
> I already wrote about that method previously [here](https://github.com/openjdk/jdk/pull/21323#discussion_r1790787304).

Sorry, missed that comment.